packages/core/core/src/factories/useSearchFactory.ts

```diff
@@ -9,21 +9,23 @@ export interface UseSearchFactoryParams<RESULT, SEARCH_PARAMS> extends FactoryPa
 export function useSearchFactory<RESULT, SEARCH_PARAMS>(
   factoryParams: UseSearchFactoryParams<RESULT, SEARCH_PARAMS>
 ) {
-  return function useSearch(id: string): UseSearch<RESULT, SEARCH_PARAMS> {
+  return function useSearch(id: string, cacheTimeToLive?: number): UseSearch<RESULT, SEARCH_PARAMS> {
     const result: Ref<RESULT> = sharedRef([], `useSearch-products-${id}`);
     const loading = sharedRef(false, `useSearch-loading-${id}`);
     const _factoryParams = configureFactoryParams(factoryParams);
     const error: Ref<UseSearchErrors> = sharedRef({
       search: null
     }, `useSearch-error-${id}`);
+    const cacheTimestamp: Ref<number> = setCacheTimestamp(`useSearch-cache-${id}`);
```

Comment:
The `setCacheTimestamp` util prevents overwriting the existing timestamp when you navigate back to the page you'd visited before. It think it could even be abbreviated from this:
```
export const setCacheTimestamp = (key: string) => {
  const { $sharedRefsMap } = useVSFContext();
  const existingTimestamp = $sharedRefsMap.get(key);
  if (existingTimestamp) return existingTimestamp;
  const timestamp = sharedRef(Date.now(), key);
  return timestamp;
};
```

to that:

```
export const setCacheTimestamp = (key: string) => {
  const { $sharedRefsMap } = useVSFContext();
  return $sharedRefsMap.get(key) || sharedRef(Date.now(), key);
};
```